specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePoolUsages.json

```diff
@@ -0,0 +1,251 @@
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/instancePools/{instancePoolName}/usages": {
+      "get": {
+        "tags": [
+          "InstancePoolUsages"
+        ],
+        "description": "Gets all instance pool usage metrics",
+        "operationId": "InstancePoolUsages_ListByInstancePool",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "name": "instancePoolName",
+            "in": "path",
+            "description": "The name of the instance pool to be retrieved.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "expandChildren",
+            "in": "query",
+            "description": "Optional request parameter to include managed instance usages within the instance pool.",
+            "required": false,
+            "type": "boolean"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the instance pool usages.",
+            "schema": {
+              "$ref": "#/definitions/UsageListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InstancePoolWrongUsageName - Request for an instance pool's usage has an unsupported usage name\n\n * 400 InstancePoolManagedInstanceInfoUnavailable - Information for managed instances inside this instance pool is not available\n\n * 404 InstancePoolNotFound - An instance pool cannot be found"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List instance pool usages.": {
+            "$ref": "./examples/ListInstancePoolUsage.json"
+          },
+          "List instance pool usages expanded with children.": {
+            "$ref": "./examples/ListInstancePoolUsageExpanded.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "UsageListResult": {
+      "description": "A list of usages.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Usage"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Usage": {
```

Comment:
do you need to add quotaPeriod and nextResetTime? going from the design doc

Reply:
they were optional I believe.

Reply:
![image](https://user-images.githubusercontent.com/8611517/59392836-f7af0d00-8d2d-11e9-9977-e9cfe3e75ebe.png)

Yeah they're not required.

Reply:
ok, as I said wasn't sure you needed them just wanted to be thorough

---
In reply to: [293155183](https://github.com/Azure/azure-rest-api-specs/pull/5894#discussion_r293155183) [](ancestors = 293155183)

Reply:
Sounds good :) appreciate it.